fdbserver/BlobManifest.actor.cpp

```diff
@@ -191,18 +191,19 @@ class BlobManifest {
 	}
 
 	// Delete all files of oldest manifest
-	static void deleteOldest(const std::vector<BlobManifestFile>& allFiles,
-	                         Reference<BackupContainerFileSystem> container) {
-		if (allFiles.empty()) {
-			return;
-		}
-		int64_t epoch = allFiles.back().epoch;
-		int64_t seqNo = allFiles.back().seqNo;
+	ACTOR static Future<Void> deleteOldest(std::vector<BlobManifestFile> allFiles,
+	                                       Reference<BackupContainerFileSystem> container) {
+		ASSERT(!allFiles.empty());
+		state int64_t epoch = allFiles.back().epoch;
+		state int64_t seqNo = allFiles.back().seqNo;
+		std::vector<Future<Void>> futures;
 		for (auto& f : allFiles) {
 			if (f.epoch == epoch && f.seqNo == seqNo) {
-				container->deleteFile(f.fileName);
+				futures.push_back(container->deleteFile(f.fileName));
 			}
 		}
+		waitForAll(futures);
```

Comment:
waitForAll just returns a future, if you want to actually wait for them to complete you have to do `wait(waitForAll(futures));`
so right now this isn't actually deleting the files